index.js

```diff
@@ -78,17 +79,22 @@ export class StuckTransactionsCanceller {
 
   async removeConfirmed (tx) {
     assert.strictEqual(typeof tx.hash, 'string')
-    await this.#store.remove(tx.hash)
+    const txs = await this.#store.list()
+    for (const _tx of txs) {
+      if (_tx.nonce === tx.nonce) {
+        await this.#store.remove(_tx.hash)
+      }
+    }
   }
 
   async cancelOlderThan (ageMs) {
     assert.strictEqual(typeof ageMs, 'number')
 
     this.#log('Checking for stuck transactions...')
     const txs = await this.#store.list()
-    const txsToCancel = txs.filter(tx => {
-      return new Date() - new Date(tx.timestamp) > ageMs
-    })
+    const txsToCancel = txs
+      .filter(tx => new Date() - new Date(tx.timestamp) > ageMs)
+      .filter(tx => !txs.find(_tx => _tx.nonce === tx.nonce && _tx.gasLimit > tx.gasLimit))
```

Comment:
Nitpick, feel free to ignore:

[Array.prototype.some()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) is a more idiomatic solution.

See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find

> If you need to find if any element satisfies the provided testing function, use [some()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some).

```suggestion
      .filter(tx => !txs.some(_tx => _tx.nonce === tx.nonce && _tx.gasLimit > tx.gasLimit))
```